### ISSUE TITLE
Add support for 'attach-mode aside'

### ISSUE THREAD

**Issue by 0xfea5** [riverwm/river#904]
Back when I was using dwm, there was a patch called 'attach aside' which spawned new windows on top of the side stack instead of the master area. 

```
Initial state
---------------------
|         |         |
|         |         |
|    A    |    B    |
|         |         |
|         |         |
---------------------


After spawning a new window
---------------------
|         |         |
|         |    C    |<- new window spawning here
|    A    |---------|
|         |         |
|         |    B    |
---------------------
```

I believe this is a very intuitive behavior and it would be nice if it was an option for it to complement 'attach top/bottom'.

I am willing to implement this, however I'd like to see some feedback before doing so.

**Comment by Leon-Plickat** [comment]
This is right now not implementable in a clean way, since the main area and window stack are concepts contained entirely in the layout generator. River itself has no idea about any of that, it just has a list of all windows.

As an alternative idea, you could create an attach mode which inserts new windows after `n` _visible_ windows. Keeping that `n` in sync with the amount of windows in the main area can be done with either a wrapper script or by having the layout generator change that setting automatically.

**Comment by 0xfea5** [comment]
> This is right now not implementable in a clean way, since the main area and window stack are concepts contained entirely in the layout generator. River itself has no idea about any of that, it just has a list of all windows.
> 
> As an alternative idea, you could create an attach mode which inserts new windows after `n` _visible_ windows. Keeping that `n` in sync with the amount of windows in the main area can be done with either a wrapper script or by having the layout generator change that setting automatically.

I see the problem here and your solution seems the only option. I'll try implement on my free time this and make a PR.

**Comment by sje30** [comment]
This sounds like a great idea.  @0xfea5 did you make a start at this?

**Comment by 0xfea5** [comment]
> This sounds like a great idea. @0xfea5 did you make a start at this?

I've spent the last month getting more familiar with Zig. I have to automate updating the number of master windows that river "sees" since this kind of information is stored inside rivertile. I will try finish it soon so it may get merged on the next release.

**Comment by johnae** [comment]
@0xfea5 For what it's worth, perhaps implementing something using https://github.com/MaxVerevkin/river-luatile would be feasible? I'm using that myself for playing around with layouts quickly and for general use as well. I believe this is also in the spirit of river (as opposed to building it into river or rivertile).

**Comment by sje30** [comment]
Hi @johnae 

thanks for highlighting river-luatile.  I'm not sure that this can solve the problem though -- as the layout manager only receives the number of windows to position; the order of the windows is set by river from what I understand.

**Comment by 0xfea5** [comment]
> @0xfea5 For what it's worth, perhaps implementing something using https://github.com/MaxVerevkin/river-luatile would be feasible? I'm using that myself for playing around with layouts quickly and for general use as well. I believe this is also in the spirit of river (as opposed to building it into river or rivertile).

As @sje30 mentioned, the order of windows is set by river. As of now you can only prepend and append to the View stack. The patch is adding a way to insert a View at an arbitrary position.

**Comment by johnae** [comment]
Thanks @sje30 and @0xfea5. I see now how that complicates things.

I suppose it might be possible to hack something together for window placement but the order wouldn't reflect that and it'd be quite messy trying to make sense of things as windows come and go.

It does seem useful to be able to manipulate the stack somehow.

**Comment by 0xfea5** [comment]
FYI this feature has been [merged](https://github.com/riverwm/river/commit/69a51cadb41443103370247ac515f1067da4b932) so I am closing this issue.

**Comment by Leon-Plickat** [comment]
I think there is still one issue that I haven't thought about earlier: The newly added command (great work btw!) Only affects either all or the currently focused output, however layouts can be generated for arbitrary outputs. So there is the possibility that in a multi-monitor setup an unfocused output needs a new layout and the layout generator sets the attach mode in response which then messes with the currently focused output.

However I think it's fine to not address this issue for now, since that responsibility will be moved out of river anyway.

**Comment by 0xfea5** [comment]
That's a problem I hadn't really thought of. A solution that comes to mind is to have something like `attach-mode-output <Output> <top|bottom|after N>`, where Output is the name of the output we want to to set the attach-mode for. The problem now, is to inform the layout generator for what output it's responsible for.

**Comment by MaxVerevkin** [comment]
Layout generators already know which output they are providing the layout for, so this is not an issue.

**Comment by 0xfea5** [comment]
It's all set then. Unless there is a better solution I can go ahead and patch this in.

**Comment by 0xfea5** [comment]
@Leon-Plickat @ifreund You may take a look before I move on
